tests/testTriggers.go

```diff
@@ -7319,6 +7319,339 @@ func TriggerMetroDR(contexts *[]*scheduler.Context, recordChan *chan *EventRecor
+				err = SetDestinationKubeConfig()
+				if err != nil {
+					log.Errorf("Failed to Set destinayion kubeconfig: %v", err)
+				}
+				nodes := node.GetStorageDriverNodes()
+				nodeIndex := rand.Intn(len(nodes))
+				log.Infof("Stop volume driver [%s] on node: [%s]", Inst().V.String(), nodes[nodeIndex].Name)
+				StopVolDriverAndWait([]node.Node{nodes[nodeIndex]})
+				log.Infof("Starting volume driver [%s] on node [%s]", Inst().V.String(), nodes[nodeIndex].Name)
+				StartVolDriverAndWait([]node.Node{nodes[nodeIndex]})
+				log.Infof("Giving a few seconds for volume driver to stabilize")
+				time.Sleep(20 * time.Second)
+				err = SetSourceKubeConfig()
+				if err != nil {
+					log.Errorf("Failed to Set source kubeconfig: %v", err)
+				}
+			})
+		}
+	}
+
+	// Validate all migrations
+	for _, mig := range allMigrations {
+			err := storkops.Instance().ValidateMigration(mig.Name, mig.Namespace, migrationRetryTimeout, migrationRetryInterval)
+			if err != nil {
+					UpdateOutcome(event, fmt.Errorf("failed to validate migration: %s in namespace %s. Error: [%v]", mig.Name, mig.Namespace, err))
+			}
+			migStats, err := asyncdr.CreateStats(mig.Name, mig.Namespace, getPXVersion(node.GetStorageNodes()[0]))
+			if err != nil {
+					UpdateOutcome(event, fmt.Errorf("Unable to create stats, getting error: %v", err))
+			}
+			dash.UpdateStats("longevity-migration-asyncdr", "stork", "migrationstatslongevity", migStats["StorkVersion"], migStats)
+	}
+	updateMetrics(*event)
+}
+
+// TriggerAsyncDRPXRestartKvdb triggers Async DR with kvdb restart
+func TriggerAsyncDRPXRestartKvdb(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer endLongevityTest()
+	startLongevityTest(AsyncDRPXRestartKvdb)
+	defer ginkgo.GinkgoRecover()
+	log.Infof("Async DR kvdb restart trigger triggered at: %v", time.Now())
+	defer ginkgo.GinkgoRecover()
+	event := &EventRecord{
+			Event: Event{
+					ID:   GenerateUUID(),
+					Type: AsyncDRPXRestartKvdb,
+			},
+			Start:   time.Now().Format(time.RFC1123),
+			Outcome: []error{},
+	}
+	defer func() {
+			event.End = time.Now().Format(time.RFC1123)
+			*recordChan <- event
+	}()
+
+	setMetrics(*event)
+
+	chaosLevel := ChaosMap[AsyncDRPXRestartKvdb]
+	var (
+			migrationNamespaces   []string
+			taskNamePrefix        = "async-dr-rkvdb"
+			allMigrations         []*storkapi.Migration
+			includeVolumesFlag    = true
+			includeResourcesFlag  = true
+			startApplicationsFlag = false
+	)
+
+	Step(fmt.Sprintf("Deploy applications for migration, with frequency: %v", chaosLevel), func() {
+
+			// Write kubeconfig files after reading from the config maps created by torpedo deploy script
+			err := asyncdr.WriteKubeconfigToFiles()
+			if err != nil {
+					log.Errorf("Failed to write kubeconfig: %v", err)
+			}
+
+			err = SetSourceKubeConfig()
+			if err != nil {
+					log.Errorf("Failed to Set source kubeconfig: %v", err)
+			}
+			UpdateOutcome(event, err)
+			for i := 0; i < Inst().GlobalScaleFactor; i++ {
```

Comment:
Do we need to reschedule applications? The Longevity test case would have already scheduled the applications, passing contexts as arguments.

Reply:
Yes, this time may be one factor but may be we need to pass multiple namespaces as async dr, so better to keep this seperate.

Reply:
As discussed, will think on it and implement in all other stork cases as well in next PR.